incubator/hnc/pkg/controllers/config_controller_test.go

```diff
@@ -0,0 +1,52 @@
+package controllers_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("HNCConfiguration", func() {
+	ctx := context.Background()
+
+	var (
+		fooName string
+		barName string
+	)
+
+	BeforeEach(func() {
+		fooName = createNS(ctx, "foo")
+		barName = createNS(ctx, "bar")
+	})
+
+	AfterEach(func() {
+		// Change current singleton back to the default value.
+		Eventually(func() error {
+			c := getHNCConfig(ctx)
+			return resetHNCConfigToDefault(ctx, c)
```

Comment:
Maybe resetHNCConfigToDefault should call getHNCConfig itself? Otherwise, that suggests that you could call it with some special config, which would kind of defeat the purpose.

Also, I'd keep this helper function (and others) in this file until it needs to be called from other files too, but that's just a nit.

Reply:
Thanks for the suggestion! Moved getHNCConfig into resetHNCConfigToDefault. It makes sense to put only shared functions in the helper file. There are not as many shared functions as before once we switch to test on Role in object_controller_test.go. I moved the ones that are not shared back to the individual files. 

I keep createNSWithLabel in the helper file even though it is only used by hierarchy_controller_test.go for now because createNSWithLabel is similar to createNS and the latter is used in many places.

Reply:
sgtm!